Bo Kyung Kim
INST447
Assignment 01: Tidying Data with Pandas

Citation:

Sadawi, N. (2016, December 28). Retrieved March 15, 2019, from https://www.youtube.com/watch?v=Q2nFweWoL2M&fbclid=IwAR22cQX9-0cmbDMZyhVf8uDtKTJTcVVYHhUTqOcJtRaVWtfz9qxuC5qso0k

School, D. (2016, April 28). Retrieved March 18, 2019, from https://www.youtube.com/watch?v=2AFGPdNn4FM


In [27]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [28]:
data = pd.read_csv("/Users/bokyungkim/Desktop/data/SAFI_openrefine.csv")

Data validation below

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131 entries, 0 to 130
Data columns (total 62 columns):
interview_date                131 non-null object
quest_no                      131 non-null int64
start                         131 non-null object
end                           131 non-null object
province                      131 non-null object
district                      131 non-null object
ward                          131 non-null object
village                       131 non-null object
years_farm                    131 non-null int64
agr_assoc                     131 non-null object
no_membrs                     131 non-null int64
_members_count                131 non-null int64
remittance_money              131 non-null object
years_liv                     131 non-null int64
parents_liv                   131 non-null object
sp_parents_liv                131 non-null object
grand_liv                     131 non-null object
sp_grand_liv                  131 non-null object


In [30]:
data.columns

Index(['interview_date', 'quest_no', 'start', 'end', 'province', 'district',
       'ward', 'village', 'years_farm', 'agr_assoc', 'no_membrs',
       '_members_count', 'remittance_money', 'years_liv', 'parents_liv',
       'sp_parents_liv', 'grand_liv', 'sp_grand_liv', 'respondent_roof_type',
       'respondent_wall_type', 'respondent_wall_type_other',
       'respondent_floor_type', 'window_type', 'buildings_in_compound',
       'rooms', 'other_buildings', 'no_plots', 'plots_count', 'water_use',
       'no_group_count', 'yes_group_count', 'no_enough_water',
       'months_no_water', 'period_use', 'exper_other', 'other_meth',
       'res_change', 'memb_assoc', 'resp_assoc', 'fees_water',
       'affect_conflicts', 'need_money', 'money_source', 'money_source_other',
       'crops_contr', 'emply_lab', 'du_labour', 'liv_owned', 'liv_owned_other',
       'liv_count', 'poultry', 'du_look_aftr_cows', 'items_owned',
       'items_owned_other', 'no_meals', 'months_lack_food',
       'no_food_m

In [31]:
#number of observations
len(data)

131

In [32]:
data.shape

(131, 62)

In [33]:
#first and last rows
data.tail()

interview_date  quest_no                     start  \
126      18-May-17       126  2017-05-18T04:13:37.000Z   
127       4-Jun-17       193  2017-06-04T09:36:20.000Z   
128       4-Jun-17       194  2017-06-04T10:13:36.000Z   
129       4-Jun-17       199  2017-06-04T10:33:55.000Z   
130       4-Jun-17       200  2017-06-04T10:52:46.000Z   

                          end province district     ward   village  \
126  2017-05-18T04:35:47.000Z   Manica   Manica  Bandula     Ruaca   
127  2017-06-04T10:13:32.000Z   Manica   Manica  Bandula     Ruaca   
128  2017-06-04T10:32:06.000Z   Manica   Manica  Bandula     Ruaca   
129  2017-06-04T10:52:22.000Z   Manica   Manica  Bandula  Chirodzo   
130  2017-06-04T11:08:13.000Z   Manica   Manica  Bandula  Chirodzo   

     years_farm agr_assoc                    ...                      \
126           5       yes                    ...                       
127          10        no                    ...                       
128           5        no                    ...                       
129          17       yes                    ...                       
130          20       yes                    ...                       

                                           items_owned  items_owned_other  \
126          ['motorcyle' ;  'radio' ;  'solar_panel']                NaN   
127  ['car' ;  'lorry' ;  'television' ;  'radio' ;...                NaN   
128  ['radio' ;  'solar_panel' ;  'solar_torch' ;  ...                NaN   
129  ['cow_cart' ;  'lorry' ;  'motorcyle' ;  'comp...                NaN   
130  ['radio' ;  'cow_plough' ;  'solar_panel' ;  '...                NaN   

    no_meals            months_lack_food  \
126        3   ['Oct' ;  'Nov' ;  'Dec']   
127        3                    ['none']   
128        3  ['Sept' ;  'Oct' ;  'Nov']   
129        3            ['Nov' ;  'Dec']   
130        3            ['Oct' ;  'Nov']   

                                    no_food_mitigation gps_Latitude  \
126                ['rely_less_food' ;  'lab_ex_food']   -19.112194   
127                                             ['na']   -19.112157   
128                                    ['lab_ex_food']   -19.112271   
129  ['rely_less_food' ;  'limit_variety' ;  'reduc...   -19.112278   
130  ['rely_less_food' ;  'restrict_adults' ;  'bor...   -19.112183   

    gps_Longitude gps_Altitude gps_Accuracy  \
126     33.483379          700          7.0   
127     33.483390          720          9.0   
128     33.483471          719         10.0   
129     33.483390          711          5.0   
130     33.483372          681         20.0   

                                    instanceID  
126  uuid:69caea81-a4e5-4e8d-83cd-9c18d8e8d965  
127  uuid:5ccc2e5a-ea90-48b5-8542-69400d5334df  
128  uuid:95c11a30-d44f-40c4-8ea8-ec34fca6bbbf  
129  uuid:ffc83162-ff24-4a87-8709-eff17abc0b3b  
130  uuid:aa77a0d7-7142-41c8-b494-483a5b68d8a7  

[5 rows x 62 columns]

Using the first citation above, I replaced the incorrect name of the village by looking at the clean data.

In [34]:
data['village'].replace('49', 'Chirodzo', inplace=True)
data['village'].replace('Ruaca-Nhamuenda', 'Ruaca', inplace=True)
data['village'].replace('Ruaca - Nhamuenda', 'Ruaca', inplace=True)
data['village'].replace('Ruca', 'Ruaca', inplace=True)
data['village'].replace('Chirdozo', 'Chirodzo', inplace=True)

Challenge Question:
The value '49' was supposed to be Chirodzo. I used SAFI_clean.csv file that was found in the link of information about that data (Data Codebook). Everything was in the same order so I looked at the village of line 50 which was "Chirodzo"

In [37]:
#months_no_water and months_lack_food for village God
#without NaN
village_god = data.loc[data['village'] == 'God', ['months_no_water','months_lack_food' ]]
no_god = village_god.dropna()
no_god

months_no_water  \
1                               ['Aug' ;  'Sept']   
6                      ['Aug' ;  'Sept' ;  'Oct']   
11            ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
12                                        ['Oct']   
14                     ['Sept' ;  'Oct' ;  'Nov']   
20                                        ['Oct']   
37                              ['Sept' ;  'Oct']   
39                              ['Sept' ;  'Oct']   
41                              ['Aug' ;  'Sept']   
77                               ['Oct' ;  'Nov']   
79                              ['Sept' ;  'Oct']   
80                     ['Sept' ;  'Oct' ;  'Nov']   
81                              ['Sept' ;  'Oct']   
82                     ['Sept' ;  'Oct' ;  'Nov']   
83                     ['Sept' ;  'Oct' ;  'Nov']   
84                               ['Oct' ;  'Nov']   
85            ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
86                               ['Jan' ;  'Dec']   
88   ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
94                              ['Sept' ;  'Oct']   
95                     ['Sept' ;  'Oct' ;  'Nov']   
102           ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
110                    ['Sept' ;  'Oct' ;  'Nov']   
117                              ['Jan' ;  'Dec']   

                                      months_lack_food  
1         ['Jan' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']  
6                                              ['Nov']  
11                                   ['Sept' ;  'Oct']  
12                          ['Sept' ;  'Oct' ;  'Nov']  
14   ['Jan' ;  'Feb' ;  'Mar' ;  'Apr' ;  'May' ;  ...  
20   ['Jan' ;  'Feb' ;  'Mar' ;  'Oct' ;  'Nov' ;  ...  
37                                             ['Nov']  
39                          ['Sept' ;  'Oct' ;  'Nov']  
41                           ['Jan' ;  'Nov' ;  'Dec']  
77                                             ['Nov']  
79                                            ['none']  
80                  ['Jan' ;  'Feb' ;  'Nov' ;  'Dec']  
81                                            ['none']  
82                                            ['none']  
83                          ['Sept' ;  'Oct' ;  'Nov']  
84                                            ['none']  
85                                             ['Nov']  
86                                             ['Nov']  
88   ['Jan' ;  'Feb' ;  'Mar' ;  'Oct' ;  'Nov' ;  ...  
94                                    ['Oct' ;  'Nov']  
95                          ['Sept' ;  'Oct' ;  'Nov']  
102                          ['Oct' ;  'Nov' ;  'Dec']  
110                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']  
117                                            ['Nov']

In [36]:
#the number of months and which months the families interviewed do not have enough food or water for village God
no_god = no_god.copy()
god_count_no_water = []
god_count_no_food = []
for index,row in no_god.iterrows():
    god_count_no_water.append(len(row['months_no_water'].split(';')))
    god_count_no_food.append(len(row['months_lack_food'].split(';')))
no_god['god_count_no_water'] = god_count_no_water
no_god['god_count_no_food'] = god_count_no_food
no_god

months_no_water  \
1                               ['Aug' ;  'Sept']   
6                      ['Aug' ;  'Sept' ;  'Oct']   
11            ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
12                                        ['Oct']   
14                     ['Sept' ;  'Oct' ;  'Nov']   
20                                        ['Oct']   
37                              ['Sept' ;  'Oct']   
39                              ['Sept' ;  'Oct']   
41                              ['Aug' ;  'Sept']   
77                               ['Oct' ;  'Nov']   
79                              ['Sept' ;  'Oct']   
80                     ['Sept' ;  'Oct' ;  'Nov']   
81                              ['Sept' ;  'Oct']   
82                     ['Sept' ;  'Oct' ;  'Nov']   
83                     ['Sept' ;  'Oct' ;  'Nov']   
84                               ['Oct' ;  'Nov']   
85            ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
86                               ['Jan' ;  'Dec']   
88   ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
94                              ['Sept' ;  'Oct']   
95                     ['Sept' ;  'Oct' ;  'Nov']   
102           ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
110                    ['Sept' ;  'Oct' ;  'Nov']   
117                              ['Jan' ;  'Dec']   

                                      months_lack_food  god_count_no_water  \
1         ['Jan' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']                   2   
6                                              ['Nov']                   3   
11                                   ['Sept' ;  'Oct']                   4   
12                          ['Sept' ;  'Oct' ;  'Nov']                   1   
14   ['Jan' ;  'Feb' ;  'Mar' ;  'Apr' ;  'May' ;  ...                   3   
20   ['Jan' ;  'Feb' ;  'Mar' ;  'Oct' ;  'Nov' ;  ...                   1   
37                                             ['Nov']                   2   
39                          ['Sept' ;  'Oct' ;  'Nov']                   2   
41                           ['Jan' ;  'Nov' ;  'Dec']                   2   
77                                             ['Nov']                   2   
79                                            ['none']                   2   
80                  ['Jan' ;  'Feb' ;  'Nov' ;  'Dec']                   3   
81                                            ['none']                   2   
82                                            ['none']                   3   
83                          ['Sept' ;  'Oct' ;  'Nov']                   3   
84                                            ['none']                   2   
85                                             ['Nov']                   4   
86                                             ['Nov']                   2   
88   ['Jan' ;  'Feb' ;  'Mar' ;  'Oct' ;  'Nov' ;  ...                   5   
94                                    ['Oct' ;  'Nov']                   2   
95                          ['Sept' ;  'Oct' ;  'Nov']                   3   
102                          ['Oct' ;  'Nov' ;  'Dec']                   4   
110                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']                   3   
117                                            ['Nov']                   2   

     god_count_no_food  
1                    5  
6                    1  
11                   2  
12                   3  
14                  11  
20                   6  
37                   1  
39                   3  
41                   3  
77                   1  
79                   1  
80                   4  
81                   1  
82                   1  
83                   3  
84                   1  
85                   1  
86                   1  
88                   6  
94                   2  
95                   3  
102                  3  
110                  4  
117                  1

In [43]:
#to save my village God data
no_god.to_csv('village_god_data.csv')

In [38]:
#months_no_water and months_lack_food for village Chirodzo
#without NaN
village_chirodzo = data.loc[data['village'] == 'Chirodzo', ['months_no_water','months_lack_food' ]]
no_chirodzo = village_chirodzo.dropna()
no_chirodzo

months_no_water  \
7                               ['Sept' ;  'Oct']   
8                                ['Oct' ;  'Nov']   
9                      ['Sept' ;  'Oct' ;  'Nov']   
33                               ['Oct' ;  'Nov']   
34                     ['Sept' ;  'Oct' ;  'Nov']   
44                              ['Sept' ;  'Oct']   
45                     ['Aug' ;  'Sept' ;  'Oct']   
46                     ['Sept' ;  'Oct' ;  'Nov']   
49                              ['Sept' ;  'Oct']   
51                     ['Sept' ;  'Oct' ;  'Nov']   
52                     ['Aug' ;  'Sept' ;  'Oct']   
53   ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
55                              ['Sept' ;  'Oct']   
59                     ['Aug' ;  'Sept' ;  'Oct']   
60                     ['Aug' ;  'Sept' ;  'Oct']   
64                     ['Sept' ;  'Oct' ;  'Nov']   
65                     ['Aug' ;  'Sept' ;  'Oct']   
66                               ['Oct' ;  'Nov']   
67                               ['Nov' ;  'Dec']   
68                               ['Nov' ;  'Dec']   
69                              ['Aug' ;  'Sept']   
125                             ['Sept' ;  'Nov']   
129                    ['Sept' ;  'Oct' ;  'Nov']   

                                      months_lack_food  
7                                              ['Jan']  
8                                     ['Jan' ;  'Dec']  
9                   ['Jan' ;  'Oct' ;  'Nov' ;  'Dec']  
33                                    ['Jan' ;  'Dec']  
34        ['Jan' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']  
44                                            ['none']  
45                          ['Sept' ;  'Oct' ;  'Nov']  
46                                            ['none']  
49   ['June' ;  'July' ;  'Aug' ;  'Sept' ;  'Oct' ...  
51                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']  
52                                             ['Nov']  
53                          ['Sept' ;  'Oct' ;  'Nov']  
55                                            ['none']  
59                                            ['none']  
60                           ['Jan' ;  'Feb' ;  'Dec']  
64                           ['Jan' ;  'Feb' ;  'Mar']  
65                                            ['none']  
66                                            ['none']  
67                                            ['none']  
68                                            ['none']  
69                                            ['none']  
125                          ['Jan' ;  'Nov' ;  'Dec']  
129                                   ['Nov' ;  'Dec']

In [39]:
#the number of months and which months the families interviewed do not have enough food or water for village Chirodzo
no_chirodzo = no_chirodzo.copy()
chirodzo_count_no_water = []
chirodzo_count_no_food = []
for index,row in no_chirodzo.iterrows():
    chirodzo_count_no_water.append(len(row['months_no_water'].split(';')))
    chirodzo_count_no_food.append(len(row['months_lack_food'].split(';')))
no_chirodzo['chirodzo_count_no_water'] = chirodzo_count_no_water
no_chirodzo['chirodzo_count_no_food'] = chirodzo_count_no_food
no_chirodzo

months_no_water  \
7                               ['Sept' ;  'Oct']   
8                                ['Oct' ;  'Nov']   
9                      ['Sept' ;  'Oct' ;  'Nov']   
33                               ['Oct' ;  'Nov']   
34                     ['Sept' ;  'Oct' ;  'Nov']   
44                              ['Sept' ;  'Oct']   
45                     ['Aug' ;  'Sept' ;  'Oct']   
46                     ['Sept' ;  'Oct' ;  'Nov']   
49                              ['Sept' ;  'Oct']   
51                     ['Sept' ;  'Oct' ;  'Nov']   
52                     ['Aug' ;  'Sept' ;  'Oct']   
53   ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
55                              ['Sept' ;  'Oct']   
59                     ['Aug' ;  'Sept' ;  'Oct']   
60                     ['Aug' ;  'Sept' ;  'Oct']   
64                     ['Sept' ;  'Oct' ;  'Nov']   
65                     ['Aug' ;  'Sept' ;  'Oct']   
66                               ['Oct' ;  'Nov']   
67                               ['Nov' ;  'Dec']   
68                               ['Nov' ;  'Dec']   
69                              ['Aug' ;  'Sept']   
125                             ['Sept' ;  'Nov']   
129                    ['Sept' ;  'Oct' ;  'Nov']   

                                      months_lack_food  \
7                                              ['Jan']   
8                                     ['Jan' ;  'Dec']   
9                   ['Jan' ;  'Oct' ;  'Nov' ;  'Dec']   
33                                    ['Jan' ;  'Dec']   
34        ['Jan' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
44                                            ['none']   
45                          ['Sept' ;  'Oct' ;  'Nov']   
46                                            ['none']   
49   ['June' ;  'July' ;  'Aug' ;  'Sept' ;  'Oct' ...   
51                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
52                                             ['Nov']   
53                          ['Sept' ;  'Oct' ;  'Nov']   
55                                            ['none']   
59                                            ['none']   
60                           ['Jan' ;  'Feb' ;  'Dec']   
64                           ['Jan' ;  'Feb' ;  'Mar']   
65                                            ['none']   
66                                            ['none']   
67                                            ['none']   
68                                            ['none']   
69                                            ['none']   
125                          ['Jan' ;  'Nov' ;  'Dec']   
129                                   ['Nov' ;  'Dec']   

     chirodzo_count_no_water  chirodzo_count_no_food  
7                          2                       1  
8                          2                       2  
9                          3                       4  
33                         2                       2  
34                         3                       5  
44                         2                       1  
45                         3                       3  
46                         3                       1  
49                         2                       7  
51                         3                       4  
52                         3                       1  
53                         5                       3  
55                         2                       1  
59                         3                       1  
60                         3                       3  
64                         3                       3  
65                         3                       1  
66                         2                       1  
67                         2                       1  
68                         2                       1  
69                         2                       1  
125                        2                       3  
129                        3                       2

In [44]:
#to save my village Chirodzo data
no_chirodzo.to_csv('village_chirodzo_data.csv')

In [40]:
#months_no_water and months_lack_food for village Ruaca
#without NaN
village_ruaca = data.loc[data['village'] == 'Ruaca', ['months_no_water','months_lack_food' ]]
no_ruaca = village_ruaca.dropna()
no_ruaca

months_no_water  \
23                                   ['Sept' ;  'Oct']   
24                                   ['Aug' ;  'Sept']   
25                                    ['Oct' ;  'Nov']   
27                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
28                          ['Sept' ;  'Oct' ;  'Nov']   
31                                   ['Sept' ;  'Oct']   
32                          ['Aug' ;  'Sept' ;  'Oct']   
70                          ['Sept' ;  'Oct' ;  'Nov']   
72                                             ['Nov']   
73                          ['Sept' ;  'Oct' ;  'Nov']   
76                                    ['Oct' ;  'Nov']   
78                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
89                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
90                          ['Sept' ;  'Oct' ;  'Nov']   
91                                   ['Sept' ;  'Oct']   
92                          ['Aug' ;  'Sept' ;  'Oct']   
93                                   ['Sept' ;  'Oct']   
96                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
97                          ['Aug' ;  'Sept' ;  'Oct']   
98                                   ['Aug' ;  'Sept']   
99                          ['Sept' ;  'Oct' ;  'Nov']   
100                                  ['Aug' ;  'Sept']   
101                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
104                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
105                                  ['Sept' ;  'Oct']   
107                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
113                         ['Sept' ;  'Oct' ;  'Nov']   
114                         ['Sept' ;  'Oct' ;  'Nov']   
115  ['Apr' ;  'May' ;  'June' ;  'July' ;  'Aug' ;...   
118       ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
119                         ['Aug' ;  'Sept' ;  'Oct']   
120  ['July' ;  'Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;...   
121                ['Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
122                                   ['Oct' ;  'Nov']   
123       ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
124                         ['Sept' ;  'Oct' ;  'Nov']   
126                         ['Sept' ;  'Oct' ;  'Nov']   
127                          ['Oct' ;  'Nov' ;  'Dec']   
128                         ['Aug' ;  'Sept' ;  'Oct']   

                                      months_lack_food  
23                                    ['Nov' ;  'Dec']  
24                           ['Jan' ;  'Feb' ;  'Oct']  
25                                            ['none']  
27                          ['Aug' ;  'Sept' ;  'Oct']  
28                                    ['Jan' ;  'Feb']  
31                                            ['none']  
32                                            ['none']  
70                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']  
72                           ['Jan' ;  'Oct' ;  'Nov']  
73                                            ['none']  
76                                            ['none']  
78                                    ['Oct' ;  'Nov']  
89   ['Jan' ;  'Aug' ;  'Sept' ;  'Oct' ;  'Nov' ; ...  
90                          ['Jan' ;  'Sept' ;  'Oct']  
91                                            ['none']  
92                          ['Aug' ;  'Sept' ;  'Oct']  
93                                    ['Oct' ;  'Nov']  
96                           ['Jan' ;  'Feb' ;  'Dec']  
97                                    ['Jan' ;  'Feb']  
98                          ['Aug' ;  'Sept' ;  'Oct']  
99                                            ['none']  
100                          ['Jan' ;  'Feb' ;  'Dec']  
101                          ['Jan' ;  'Feb' ;  'Dec']  
104                                           ['none']  
105                                           ['none']  
107       ['Jan' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']  
113                                           ['none']  
114                                   ['Jan' ;  'Dec']  
115                         ['Sept' ;  'Oct' ;  'Nov']  
118     

In [41]:
#the number of months and which months the families interviewed do not have enough food or water for village Ruaca
no_ruaca = no_ruaca.copy()
ruaca_count_no_water = []
ruaca_count_no_food = []
for index,row in no_ruaca.iterrows():
    ruaca_count_no_water.append(len(row['months_no_water'].split(';')))
    ruaca_count_no_food.append(len(row['months_lack_food'].split(';')))
no_ruaca['ruaca_count_no_water'] = ruaca_count_no_water
no_ruaca['ruaca_count_no_food'] = ruaca_count_no_food
no_ruaca

months_no_water  \
23                                   ['Sept' ;  'Oct']   
24                                   ['Aug' ;  'Sept']   
25                                    ['Oct' ;  'Nov']   
27                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
28                          ['Sept' ;  'Oct' ;  'Nov']   
31                                   ['Sept' ;  'Oct']   
32                          ['Aug' ;  'Sept' ;  'Oct']   
70                          ['Sept' ;  'Oct' ;  'Nov']   
72                                             ['Nov']   
73                          ['Sept' ;  'Oct' ;  'Nov']   
76                                    ['Oct' ;  'Nov']   
78                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
89                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
90                          ['Sept' ;  'Oct' ;  'Nov']   
91                                   ['Sept' ;  'Oct']   
92                          ['Aug' ;  'Sept' ;  'Oct']   
93                                   ['Sept' ;  'Oct']   
96                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
97                          ['Aug' ;  'Sept' ;  'Oct']   
98                                   ['Aug' ;  'Sept']   
99                          ['Sept' ;  'Oct' ;  'Nov']   
100                                  ['Aug' ;  'Sept']   
101                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
104                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
105                                  ['Sept' ;  'Oct']   
107                ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']   
113                         ['Sept' ;  'Oct' ;  'Nov']   
114                         ['Sept' ;  'Oct' ;  'Nov']   
115  ['Apr' ;  'May' ;  'June' ;  'July' ;  'Aug' ;...   
118       ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
119                         ['Aug' ;  'Sept' ;  'Oct']   
120  ['July' ;  'Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;...   
121                ['Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
122                                   ['Oct' ;  'Nov']   
123       ['Aug' ;  'Sept' ;  'Oct' ;  'Nov' ;  'Dec']   
124                         ['Sept' ;  'Oct' ;  'Nov']   
126                         ['Sept' ;  'Oct' ;  'Nov']   
127                          ['Oct' ;  'Nov' ;  'Dec']   
128                         ['Aug' ;  'Sept' ;  'Oct']   

                                      months_lack_food  ruaca_count_no_water  \
23                                    ['Nov' ;  'Dec']                     2   
24                           ['Jan' ;  'Feb' ;  'Oct']                     2   
25                                            ['none']                     2   
27                          ['Aug' ;  'Sept' ;  'Oct']                     4   
28                                    ['Jan' ;  'Feb']                     3   
31                                            ['none']                     2   
32                                            ['none']                     3   
70                 ['Aug' ;  'Sept' ;  'Oct' ;  'Nov']                     3   
72                           ['Jan' ;  'Oct' ;  'Nov']                     1   
73                                            ['none']                     3   
76                                            ['none']                     2   
78                                    ['Oct' ;  'Nov']                     4   
89   ['Jan' ;  'Aug' ;  'Sept' ;  'Oct' ;  'Nov' ; ...                     4   
90                          ['Jan' ;  'Sept' ;  'Oct']                     3   
91                                            ['none']                     2   
92                          ['Aug' ;  'Sept' ;  'Oct']                     3   
93                                    ['Oct' ;  'Nov']                     2   
96                           ['Jan' ;  'Feb' ;  'Dec']                     4   
97                                    ['Jan' ;  'Feb']                     3   
98                          ['Aug' ;  'Sept' ;  'Oct']                     2   
99                                    

In [45]:
#to save my village Ruaca data
no_ruaca.to_csv('village_ruaca_data.csv')

Questions:
1. There are total of 131 observations. Observation is the list of given data, so for this one is a data of an excerpt of 3 villages on interview dataes between November 2016 and Novermber 2017. The level of analysis was appropriate to understand, however, it could've been improved. Without the information link given, I would've have trouble understanding some of the columns.

2. 17 unique types of possessions were identified.

3. 13 different strategies were used to mitigate the effect of not having enough food.

4. There was somewhat significant measurable impact of being a member of an agricultural association and the number of months without water. If you see the number down below, you can see that the numbers match pretty similar to member of an agricultural assoication, and number of months without water for Chirodzo and God. However, numbers of months without water is pretty high compare to member of an agricultural association. 

In [13]:
#yes under no_enough_water had months of no enough water therefore, I used no_enough_water column to compare with agr_assoc
data['no_enough_water']=data['no_enough_water'].replace('yes', int(1))
data['no_enough_water']=data['no_enough_water'].replace('no', int(0))
data.groupby('village').no_enough_water.sum()

village
Chirodzo    23
God         24
Ruaca       39
Name: no_enough_water, dtype: int64

In [14]:
data['agr_assoc']=data['agr_assoc'].replace('yes', int(1))
data['agr_assoc']=data['agr_assoc'].replace('no', int(0))
data.groupby('village').agr_assoc.sum()

village
Chirodzo    28
God         25
Ruaca       23
Name: agr_assoc, dtype: int64

Question (cont.):
5. It was most interesting to look into different parts of the data because this data contained many different information. For example, household features, agricultural practices, assets and details about the household members. However, it contained few errors.